documentation/general/torn-sdk.md

```diff
@@ -0,0 +1,142 @@
+
+This also hits any customer that uses a preview version of .NET SDK. These inherently represent a torn SDK state because they almost never match the compiler in Visual Studio This results in blockers for big teams like Bing from testing out our previews.
+
+## Goals
+
+This design has a number of goals:
+
+1. The `msbuild` and `dotnet build` build experience should be equivalent.
+1. The Visual Studio Design time experience is independent of the .NET SDK installed.
+1. To make explicit that it is okay, and even expected, that the design time and command line build experiences can differ when the SDK is in a torn state.
+
+## MSBuild using .NET SDK Compiler
+
+The .NET SDK will start producing a package named Microsoft.NETSdk.Compiler.Roslyn. This will contain a .NET Framework version of the Roslyn compiler that matches .NET Core version. This will be published as a part of the .NET SDK release process meaning it's available for all publicly available .NET SDKs.
+
+The .NET SDK has the capability to [detect a torn state][pr-detect-torn-state]. When this is detected the matching Microsoft.NETSDK.Compiler.Roslyn package for this version of the .NET SDK will be downloaded via `<PackageDownload>`. Then the `$(RoslynTargetsPath)` will be reset to point into the package contents. This will cause MSBuild to load the Roslyn compiler from that location vs. the Visual Studio location.
+
+One downside to this approach is that it is possible to end up with two VBCSCompiler server processes. Consider that when a solution has a mix of .NET SDK style projects and legacy projects then in a torn state the .NET SDK projects will use the .NET SDK compiler will legacy projects will use the Visual Studio compiler. While not a desirable outcome, it is a correct one. Customers who wish to only have one VBCSCompiler should correct the torn state in their build tools.
+
+## Visual Studio using Visual Studio Analyzers
+
+### .NET SDK in box analyzers dual insert
+
+Analyzers which ship in the .NET SDK box will change to having a copy checked into Visual Studio. When .NET SDK based projects are loaded at design time, the Visual Studio copy of the analyzer will be loaded. Roslyn already understands how to prefer Visual Studio copies of analyzers. That work will need to be extended a bit but that is pretty straight forward code.
+
+This approach enables us to take actions like NGEN or R2R analyzers inside of Visual Studio. This is a long standing request from the Visual Studio perf team but very hard to satisfy in our current
+
+This approach is already taken by [the Razor generator][code-razor-vs-load]. This work was done for other reliability reasons but turned out working well for this scenario. There is initial upfront work to get the Visual Studio copy loading in design time builds but it has virtually zero maintanence cost.
```

Comment:
You say virtually zero maintenance cost. Is that because Razor is already inserting into VS so including their generator was low marginal cost. What about teams like aspnet and windows desktop that don't have separate insertions today? Wouldn't this be adding additional insertion management to them separate from the SDK insertion?

Reply:
>  Is that because Razor is already inserting into VS so including their generator was low marginal cost

Basically yes. Once we had done the work insert the generator it just happens automatically as part of the Razor insertion. 

> What about teams like aspnet and windows desktop that don't have separate insertions today?

My expectation is that we'll settle on a design where .NET SDK insertions end up inserting all of the inbox analyzers. The copy used in the .NET SDK is the same that will be used in VS. Hence the most natural way to handle this is to insert these at the time the .NET SDK inserts into VS. That has the added bonus that we get VS perf runs on the experience as a whole.